### PR TITLE
feat(wallet): Add Transak on-ramp provider

### PR DESCRIPTION
**Pull request by muliswilliam**
Resolves https://github.com/brave/brave-browser/issues/26737

Note: Some icons in this build may be missing, they will be available when the PR is merged since they have already been added in this PR: https://github.com/brave/token-lists/pull/33

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Buy Flow
https://user-images.githubusercontent.com/48117473/201928138-5fbd4258-4a06-435c-bb83-0a425898da2a.mov

### Buy flow in onboarding
https://user-images.githubusercontent.com/48117473/201969496-f3bb51db-4607-491d-a65b-f3dbadfe31ad.mov

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_constants.h
Comment (kdenhartog) [inline components/brave_wallet/browser/brave_wallet_constants.h:40]: Hardcoding this API key is not a great idea as it allows anyone to impersonate requests as Brave. Is there any other options for authentication here such as a PKCE Auth flow?

Cc @fmarier who was recently looking at this class of issues who may have a better understanding off the top of his head.

File: components/brave_wallet/browser/brave_wallet_constants.cc
Comment (kdenhartog) [inline components/brave_wallet/browser/brave_wallet_constants.cc:256]: Nit that's best addressed in a separate issue - I'm thinking we're doing this commonly enough that it makes sense for us to abstract these lists into a separate repository that we can update with a component updater. Cc @yrliou whats your take on this?

File: components/brave_wallet/browser/asset_ratio_service.cc
Comment (kdenhartog) [inline components/brave_wallet/browser/asset_ratio_service.cc:237]: Yeah looks like it would be easy for someone to impersonate us with this API key. I'll need to look at what privileges are granted to this token before I can approve this.

**Review by kdenhartog — CHANGES_REQUESTED** [review]
Just the hardcoding of API keys is the blocking issue here. I'll take a look at it in the next day or two.

File: components/brave_wallet/browser/brave_wallet_constants.cc
Comment (muliswilliam) [reply components/brave_wallet/browser/brave_wallet_constants.cc:256]: @kdenhartog We are already exploring this.

File: components/brave_wallet/browser/brave_wallet_constants.h
Comment (muliswilliam) [reply components/brave_wallet/browser/brave_wallet_constants.h:40]: @kdenhartog The `apiKey` will still be shown to the user when the buy url is generated and opened in a new tab. I am not sure there is a way to avoid this from happening.

File: components/brave_wallet/common/brave_wallet.mojom
Comment (fmarier) [inline components/brave_wallet/common/brave_wallet.mojom:528]: Is this used anywhere?

File: components/brave_wallet/browser/asset_ratio_service.cc
Comment (fmarier) [inline components/brave_wallet/browser/asset_ratio_service.cc:238]: I'd suggest putting this parameter last so that the other parameters come first. This one is the least important from the user's point of view.

Comment (muliswilliam) [reply components/brave_wallet/browser/asset_ratio_service.cc]: Done 9001a15265a7c64dc8d73eefe81db5786c41dda7

File: components/brave_wallet/browser/asset_ratio_service.cc
Comment (fmarier) [reply components/brave_wallet/browser/asset_ratio_service.cc]: According to Transak, the only privileges associated with that API key are the ones related to what we're doing with it (i.e. buying crypto for a given address).

**Review by fmarier — APPROVED** [review]

File: components/brave_wallet/browser/asset_ratio_service.cc
Comment (kdenhartog) [reply components/brave_wallet/browser/asset_ratio_service.cc]: In that case, I think this one is good to go. Thanks for checking

**Review by kdenhartog — APPROVED** [review]
thanks @fmarier for looking into the API key usage. This one seems G2G for security review

**Review by Douglashdaniel — APPROVED** [review]
++

**Review by cypt4 — APPROVED** [review]
LGTM

**Comment by srirambv** [comment]
Verification passed on

Brave | 1.47.84 Chromium: 108.0.5359.48 (Official Build) nightly (64-bit)
-- | --
Revision | `18ceeca0d99318e70c00d2e04d88aa55488b5c63-refs/branch-heads/5359@{#854}`
OS | Linux


- Verified steps from PR
- Verified Transak on-ramp provider is shown during onboarding for Buy Crypto
- Verified Transak on-ramp provider is shown in Buy widget for tokens on Solana Network and Ethereum/EVM Network
- Verified Transak on-ramp provider is not shown for Filecoin

https://user-images.githubusercontent.com/17010094/202736891-34b9e7ed-bccf-458c-89c2-413a1464f710.mp4

**Pull request merged**
